packages/flutter/lib/src/foundation/print.dart

```diff
@@ -31,6 +31,14 @@ typedef DebugPrintCallback = void Function(String? message, { int? wrapWidth });
 /// Prints a message to the console, which you can access using the "flutter"
 /// tool's "logs" command ("flutter logs").
 ///
+/// `debugPrint` logs to console even in [release mode](https://docs.flutter.dev/testing/build-modes#release).
+/// To disable logging from [debugPrint] in release mode, set it to a no-op:
+/// ```dart
+/// if(kReleaseMode) {
+///   debugPrint = (_, {int? wrapWidth}) {};
+/// }
+/// ```
```

Comment:
I suspect that with this code the callsites are not actually tree shaken out and the strings (plus all the machinery to generate those, which could be huge if e.g. `runtimeType` is used) remains in the release binary.

The better advice to give here is that calls to debugPrint should only happen from a debug context (e.g. from within a `if (kDebugMode) { }` block or within an assert). In general, that is the convention in the framework: methods starting with `debug*` should only be invoked from a debug context. `debugPrint` is no exception there.

Reply:
> I suspect that with this code the callsites are not actually tree shaken out and the strings

Interesting, I wasn't aware of that

Reply:
I don't think so